modules/apps/site/site-cms-site-initializer/src/main/java/com/liferay/site/cms/site/initializer/internal/display/context/ViewVersionHistoryDisplayContext.java

```diff
@@ -14,56 +14,70 @@
 import com.liferay.petra.string.StringPool;
 import com.liferay.portal.kernel.exception.PortalException;
 import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.HashMapBuilder;
 import com.liferay.portal.kernel.util.ParamUtil;
 import com.liferay.portal.kernel.util.WebKeys;
 
 import jakarta.servlet.http.HttpServletRequest;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
 /**
  * @author Mikel Lorza
  */
 public class ViewVersionHistoryDisplayContext {
 
 	public ViewVersionHistoryDisplayContext(
-		HttpServletRequest httpServletRequest,
-		ObjectDefinitionLocalService objectDefinitionLocalService,
-		ObjectEntryService objectEntryService) {
+			HttpServletRequest httpServletRequest,
+			ObjectDefinitionLocalService objectDefinitionLocalService,
+			ObjectEntryService objectEntryService)
+		throws PortalException {
 
 		_httpServletRequest = httpServletRequest;
 		_objectDefinitionLocalService = objectDefinitionLocalService;
-		_objectEntryService = objectEntryService;
+
+		_objectEntry = objectEntryService.getObjectEntry(
+			ParamUtil.getLong(httpServletRequest, "objectEntryId"));
```

Comment:
In general, it is best to make constructors as dumb as possible. Can you fetch the object from outside the constructor and pass it in as another constructor parameter instead?